addons/website_event_booth/static/src/interactions/booth_register.js

```diff
@@ -0,0 +1,220 @@
+                }
+            }
+            const unavailableBoothAlertEl = this.el.querySelector(".o_wbooth_unavailable_booth_alert");
+            unavailableBoothAlertEl.classList.remove("d-none");
+            return false;
+        }
+        return true;
+    }
+
+    countSelectedBooths() {
+        return this.el.querySelectorAll(".form-check > input[type='checkbox']:checked").length;
+    }
+
+    updateBoothsList() {
+        const boothsElem = this.el.querySelector('.o_wbooth_booths');
+        boothsElem.replaceChildren(renderToFragment('event_booth_checkbox_list', {
+            'event_booth_ids': this.boothCache[this.activeBoothCategoryId],
+            'selected_booth_ids': this.isFirstRender ? this.selectedBoothIds : [],
+        }));
+        this.isFirstRender = false;
+    }
+
+    /**
+     * Check if the confirmation form is valid by testing each of its inputs
+     *
+     * @param formEl
+     * @return {boolean} - true if no errors else false
+     */
+    checkConfirmationForm(formEl) {
+        const formControlEls = formEl.querySelectorAll(".form-control");
+        const formErrors = [];
+        for (const formControlEl of formControlEls) {
+            formControlEl.classList.remove("is-invalid");
+            if (!formControlEl.checkValidity()) {
+                formControlEl.classList.add("is-invalid");
+                formErrors.push('invalidFormInputs');
+            }
+        }
+        this.updateErrorDisplay(formErrors);
+        return formErrors.length === 0;
+    }
+
+    showBoothCategoryDescription() {
+        const boothCategoryDescriptionEls = this.el.querySelectorAll(".o_wbooth_booth_category_description");
+        for (const boothCategoryDescriptionEl of boothCategoryDescriptionEls) {
+            boothCategoryDescriptionEl.classList.add("d-none");
+        }
+        const activeBoothEl = this.el.querySelector("#o_wbooth_booth_description_" + this.activeBoothCategoryId);
+        activeBoothEl.classList.remove("d-none");
+    }
+
+    /**
+     * Display the errors with a custom message when confirming
+     * the registration if there is any.
+     *
+     * @param errors
+     */
+    updateErrorDisplay(errors) {
+        const errorSectionEl = this.el.querySelector(".o_wbooth_registration_error_section");
+        errorSectionEl.classList.toggle("d-none", !errors.length);
+
+        const errorMessages = [];
+        if (errors.includes('invalidFormInputs')) {
+            errorMessages.push(_t("Please fill out the form correctly."));
+        }
+        if (errors.includes('boothError')) {
+            errorMessages.push(_t("Booth registration failed."));
+        }
+        if (errors.includes('boothCategoryError')) {
+            errorMessages.push(_t("The booth category doesn't exist."));
+        }
+
+        const errorMessageEl = this.el.querySelector(".o_wbooth_registration_error_message");
+        errorMessageEl.textContent = errorMessages.join(" ");
```

Comment:
Can't this be a `t-out` ?

Reply:
I'm not sure since we trigger an event for the change after this.